### ISSUE TITLE
CLI and Config Redux

### ISSUE THREAD

**Issue by scottietremendous** [concourse/concourse#6571]
### TLDR

- Using a single configuration file to start up the Concourse binary.

### Problem

- Currently the only way to configure the concourse binary is through flags or environment variables. This makes it hard to keep a consistent state of values for each of the fields and to properly maintain a version history of each deployment state.

### Proposed Solution

- It would be more efficient to use a single configuration file in order to start up the Concourse binary. This means that there will be only one configuration file in YAML format for both the web and worker nodes.

### Outcomes

- Having a simple config file will help with clarity of exactly what is being set in a deployment without needing to read through every flag/env var that you need to pass in or set when you are starting up concourse.
- This also lowers the effort involved in creating the BOSH release and Helm chart. This would help the Concourse team release faster and easier than we currently do.
- Improves installation and configuration experience, and means that documenting these processes is easier as well.

### Current Status

- In progress

### References

- #6434
